### ISSUE TITLE
kitty-client always focuses client0

### ISSUE THREAD

**Issue by jtrees** [mawww/kakoune#4055]
### Steps 
1. In kitty terminal A: `kak -s foo`
2. In kitty terminal B: `kak -c foo`
3. In kakoune on any terminal: `:kitty-focus client1`

### Outcome 
Kitty terminal A is focused.

### Expected 
Kitty terminal B should be focused.

### Additional Info
- Kakoune v2020.09.01-217-g14f7d263
- kitty 0.19.2
- xmonad 0.15 (window manager)
- Linux 5.7.1

**Comment by Parasrah** [comment]
A little late but I override the default kitty config in kakoune with my own, and `kitty-focus client1` works for me

[kitty.kak](https://github.com/parasrah/kitty.kak)
[kitty conf](https://github.com/Parasrah/dotfiles/blob/master/kitty/kitty.conf)

Not sure if this is even still a problem in latest kakoune, but figured I'd share in case it helps

**Comment by jtrees** [comment]
Thanks. I tried out the plugin you posted but I still get the same behaviour. Even get it across different machines with different distros. That makes me wonder if it might be xmonad related...

**Comment by Parasrah** [comment]
I think an important part of getting it to work was my kitty config as well, with the `listen_on` and whatnot

**Comment by jtrees** [comment]
Yeah, I tried it with the lines from your "remote control" config section.

**Comment by jtrees** [comment]
The behaviour has changed for me (it's still wrong though) so I updated the description.

**Comment by jtrees** [comment]
I have come to the conclusion that I was _doing it wrong_. Turns out I running several instances of Kitty but the windows actually only communicate with each other within the same instance. If I create all OS-windows as part of the same kitty instance ([docs](https://sw.kovidgoyal.net/kitty/invocation/#cmdoption-kitty-single-instance)), everything works as expected:

```sh
# Launch kitty like this.
kitty --single-instance
```

I do think Kakoune's behaviour here is confusing though. It shouldn't blindly use `KITTY_WINDOW_ID`s if they come from different kitty instances because that results in undefined behaviour. The `KITTY_PID` env var can be used to check this.
